sdk/formrecognizer/Azure.AI.FormRecognizer/tests/DocumentModelAdministrationClient/DocumentModelAdministrationLiveTests.cs

```diff
@@ -176,170 +176,237 @@ public void BuildModelError()
+                }
+            }
 
             Assert.IsTrue(operation.HasValue);
 
-            DocumentModelDetails copiedModel = operation.Value;
-
-            ValidateDocumentModelDetails(copiedModel);
-            Assert.AreEqual(targetAuth.TargetModelId, copiedModel.ModelId);
-            Assert.AreNotEqual(trainedModel.ModelId, copiedModel.ModelId);
+            DocumentModelDetails sourceModel = disposableModel.Value;
+            DocumentModelDetails model = operation.Value;
 
-            Assert.AreEqual(1, copiedModel.DocumentTypes.Count);
-            Assert.IsTrue(copiedModel.DocumentTypes.ContainsKey(modelId));
+            Assert.AreEqual(modelId, model.ModelId);
+            Assert.AreEqual(description, model.Description);
+            Assert.AreEqual(ServiceVersionString, model.ApiVersion);
+            Assert.Greater(model.CreatedOn, startTime);
+            Assert.Greater(model.ExpiresOn, model.CreatedOn);
 
-            DocumentTypeDetails documentType = copiedModel.DocumentTypes[modelId];
+            CollectionAssert.AreEquivalent(tags, model.Tags);
 
-            Assert.AreEqual(DocumentBuildMode.Template, documentType.BuildMode);
+            AssertDocumentTypeDictionariesAreEquivalent(sourceModel.DocumentTypes, model.DocumentTypes);
         }
 
+        #endregion Copy
+
+        #region Compose
+
         [RecordedTest]
-        public async Task CopyModelToWithTags()
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task ComposeDocumentModel(bool useTokenCredential)
         {
-            var sourceClient = CreateDocumentModelAdministrationClient();
-            var targetClient = CreateDocumentModelAdministrationClient();
+            var client = CreateDocumentModelAdministrationClient(useTokenCredential);
+            var componentModelIds = new string[] { Recording.GenerateId(), Recording.GenerateId() };
             var modelId = Recording.GenerateId();
+            var description = $"This model was generated by a .NET test: {nameof(ComposeDocumentModel)}";
+            var tags = new Dictionary<string, string>() { { "tag1", "value1" }, { "tag2", "value2" } };
+            var startTime = Recording.UtcNow;
+
+            await using var disposableModel0 = await BuildDisposableDocumentModelAsync(componentModelIds[0], ContainerType.Singleforms);
+            await using var disposableModel1 = await BuildDisposableDocumentModelAsync(componentModelIds[1], ContainerType.SelectionMarks);
+
+            ComposeDocumentModelOperation operation = null;
+
+            try
+            {
+                operation = await client.ComposeDocumentModelAsync(WaitUntil.Completed, componentModelIds, modelId, description, tags);
+            }
+            finally
+            {
+                if (operation != null && operation.HasValue)
+                {
+                    await client.DeleteDocumentModelAsync(modelId);
+                }
+            }
```

Comment:
We also had leak on the "compose" method. Similarly to copy, we were doing:
1. Create models A and B.
2. Compose models A and B into a new model C.
3. Do some validation.
4. Delete models A and B (forgot about C).

Reply:
Do you mean memory leak here? If you don't delete a model - it would just continue to be associated with the cognitive resource. Not sure I understand what leak means in this case.

Reply:
I'm calling it a "model leak", and it's exactly as you described:
> If you don't delete a model - it would just continue to be associated with the cognitive resource

When running in our live tests pipeline it doesn't matter much since we'll be deleting the Form Recognizer resource at the end of the test run.

However, when running live tests locally, these models eventually build up over time in the resource since it won't be deleted. When it reaches the maximum quota of models per resource, tests start failing and the accumulated models must be deleted manually.

Also, in the past there were cases when we temporarily had to use static resources in the live test pipeline (usually a cross-language resource). Having model leaks in daily tests running for 4 languages makes our resource reach the quota in just a few days.